### ISSUE TITLE
BUG: we need a nice error message for key word version errors

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#2552]
# problem
the user gets a python error if they open data with keyword version greater than the current software version. This can happen if a user is testing dev 3.3 and then steps back to master 3.2

# error message 
```
An error has occured while executing Python code:


Traceback (most recent call last):
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\gui\widgets\dock.py", line 971, in get_layers
    self.keyword_io.update_keywords(layer, {'title': name})
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\utilities\keyword_io.py", line 180, in update_keywords
    existing_keywords = self.read_keywords(layer)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\utilities\keyword_io.py", line 121, in read_keywords
    keywords = read_file_keywords(source, keyword)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\utilities\utilities.py", line 328, in read_file_keywords
    raise KeywordNotFoundError(message)
KeywordNotFoundError: Keyword retrieval failed for C:/AIFDR/AIFDR II - SDA/05 Test Data/DKI Jakarta/01 Hazard Data/Jakarta_Flood_HKV_WGS84.keywords (None) 
 'NoneType' object has no attribute 'split'


Python version:
2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]


QGIS version:
2.8.3-Wien Wien, 313a6bc

Python path: ['C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\\processing', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python', u'C:/Users/AIFDR_ADMIN/.qgis2/python', u'C:/Users/AIFDR_ADMIN/.qgis2/python/plugins', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\matplotlib-1.3.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\nose-1.3.3-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\tornado-4.0.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\backports.ssl_match_hostname-3.4.0.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\certifi-14.05.14-py2.7.egg', 'c:\\osgeo4~1\\apps\\python27\\lib\\site-packages\\python_dateutil-2.1-py2.7.egg', 'c:\\osgeo4~1\\apps\\python27\\lib\\site-packages\\six-1.3.0-py2.7.egg', 'C:\\Program Files (x86)\\ArcGIS\\Desktop10.2\\bin', 'C:\\Program Files (x86)\\ArcGIS\\Desktop10.2\\arcpy', 'C:\\Program Files (x86)\\ArcGIS\\Desktop10.2\\ArcToolbox', 'C:\\PROGRA~1\\QGISWI~1\\bin\\python27.zip', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\DLLs', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\plat-win', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\lib-tk', 'C:\\PROGRA~1\\QGISWI~1\\bin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\PIL', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\jinja2-2.7.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\markupsafe-0.23-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\pytz-2012j-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Pythonwin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Shapely-1.2.18-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\six-1.3.0-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\wx-2.8-msw-unicode', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlrd-0.9.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlwt-0.7.5-py2.7.egg', u'C:/Users/AIFDR_ADMIN/.qgis2//python', 'C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\\inasafe', 'C:\\Users\\AIFDR_ADMIN\\.qgis2\\python\\plugins\\inasafe\\safe_extras\\parameters', 'C:\\Users\\AIFDR_ADMIN\\.qgis2\\python\\plugins', 'C:\\Users\\AIFDR_ADMIN\\.qgis2\\python\\plugins\\inasafe\\safe_extras', 'C:\\PROGRA~1\\QGISWI~1\\apps\\qgis-ltr\\python\\plugins\\fTools\\tools']
```

# cc
@samnawi - thanks for discovering this

# proposed solution
make it happy

**Comment by Charlotte-Morgan** [comment]
I also get the following error message when I try to assign keywords to the data in 3.2.2 (master) once the data have been viewed in 3.3 (dev)
```
An error has occured while executing Python code:


Traceback (most recent call last):
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\gui\tools\wizard_dialog.py", line 3801, in on_pbnNext_released
    self.save_current_keywords()
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\gui\tools\wizard_dialog.py", line 4416, in save_current_keywords
    layer=self.layer, keywords=current_keywords)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\utilities\keyword_io.py", line 161, in write_keywords
    write_keywords_to_file(source, keywords)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\utilities\utilities.py", line 360, in write_keywords_to_file
    safe_write_keywords(keywords, filename)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\storage\utilities.py", line 171, in write_keywords
    existing_keywords = read_keywords(filename, all_blocks=True)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\storage\utilities.py", line 269, in read_keywords
    metadata = read_iso_metadata(keyword_filename)
  File "C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\inasafe\safe\storage\metadata_utilities.py", line 215, in read_iso_metadata
    metadata = {'keywords': keyword_element.text.split('\n')}
AttributeError: 'NoneType' object has no attribute 'split'


Python version:
2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]


QGIS version:
2.8.3-Wien Wien, 313a6bc

Python path: ['C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\\processing', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python', u'C:/Users/AIFDR_ADMIN/.qgis2/python', u'C:/Users/AIFDR_ADMIN/.qgis2/python/plugins', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\matplotlib-1.3.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\nose-1.3.3-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\tornado-4.0.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\backports.ssl_match_hostname-3.4.0.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\certifi-14.05.14-py2.7.egg', 'c:\\osgeo4~1\\apps\\python27\\lib\\site-packages\\python_dateutil-2.1-py2.7.egg', 'c:\\osgeo4~1\\apps\\python27\\lib\\site-packages\\six-1.3.0-py2.7.egg', 'C:\\Program Files (x86)\\ArcGIS\\Desktop10.2\\bin', 'C:\\Program Files (x86)\\ArcGIS\\Desktop10.2\\arcpy', 'C:\\Program Files (x86)\\ArcGIS\\Desktop10.2\\ArcToolbox', 'C:\\PROGRA~1\\QGISWI~1\\bin\\python27.zip', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\DLLs', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\plat-win', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\lib-tk', 'C:\\PROGRA~1\\QGISWI~1\\bin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\PIL', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\jinja2-2.7.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\markupsafe-0.23-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\pytz-2012j-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Pythonwin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Shapely-1.2.18-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\six-1.3.0-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\wx-2.8-msw-unicode', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlrd-0.9.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlwt-0.7.5-py2.7.egg', u'C:/Users/AIFDR_ADMIN/.qgis2//python', 'C:/Users/AIFDR_ADMIN/.qgis2/python/plugins\\inasafe', 'C:\\Users\\AIFDR_ADMIN\\.qgis2\\python\\plugins\\inasafe\\safe_extras\\parameters', 'C:\\Users\\AIFDR_ADMIN\\.qgis2\\python\\plugins', 'C:\\Users\\AIFDR_ADMIN\\.qgis2\\python\\plugins\\inasafe\\safe_extras', 'C:\\PROGRA~1\\QGISWI~1\\apps\\qgis-ltr\\python\\plugins\\fTools\\tools']
```

**Comment by ismailsunni** [comment]
Hi @Charlotte-Morgan and @samnawi 
After looking at this issue, I found it will be more general problem, the layer has invalid xml file. It will raise the same problem.
So, I made a fix in #2555. Short summary, it will assume that the keyword is invalid for inasafe v3.2.x if found invalid xml file.

**Comment by ismailsunni** [comment]
Closing this, the PR is merged to master.
